runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -41,11 +44,27 @@ public StreamViewSMRAdapter(CorfuRuntime runtime,
         this.streamView = streamView;
     }
 
+    private List<SMREntry> dataAndCheckpointMapper(ILogData logData) {
+        if (logData.hasCheckpointMetadata()) {
+            // This is a CHECKPOINT record.  Extract the SMREntries, if any.
+            CheckpointEntry cp = (CheckpointEntry) logData.getPayload(runtime);
```

Comment:
I see the error about the NPE. Why not restore the entry here instead of modifyingAddressSpaceView?

so

``cp.getSMREntries().forEach(x -> x.setEntry(cp.getEntry()))```